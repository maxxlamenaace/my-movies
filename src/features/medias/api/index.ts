import { getMediaCredits } from './get-media-credits/get-media-credits';
import { getMediaDetails } from './get-media-details/get-media-details';
import { getMediaGenres } from './get-media-genres/get-media-genres';
import { getMediaImages } from './get-media-images/get-media-images';
import { getMediaRecommendations } from './get-media-recommendations/get-media-recommendations';
import { getMediaVideos } from './get-media-videos/get-media-videos';
import { getMediasList } from './get-medias-list/get-medias-list';

export {
  getMediaCredits,
  getMediaDetails,
  getMediaGenres,
  getMediaImages,
  getMediaRecommendations,
  getMediaVideos,
  getMediasList,
};
