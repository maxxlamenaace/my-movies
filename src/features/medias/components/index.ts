import CircularRate from './circular-rate/CircularRate';
import MediasAutoSwiper from './medias-auto-swiper/MediasAutoSwiper';
import MediasList from './medias-list/MediasList';
import MediaItem from './media-item/MediaItem';
import MediaBackgroundHeader from './media-background-header/MediaBackgroundHeader';
import MediaInfos from './media-infos/MediaInfos';
import CastSlide from './cast-slide/CastSlide';
import MediaVideosSlide from './media-videos-slide/MediaVideosSlide';
import MediaVideoItem from './media-video-item/MediaVideoItem';
import MediaImagesSlide from './media-images-slide/MediaImagesSlide';
import MediasGrid from './medias-grid/MediasGrid';
import FavoriteItem from './favorite-item/FavoriteItem';
import FavoritesGrid from './favorites-grid/FavoritesGrid';

export {
  CircularRate,
  MediasAutoSwiper,
  MediasList,
  MediaItem,
  MediaBackgroundHeader,
  MediaInfos,
  CastSlide,
  MediaVideosSlide,
  MediaVideoItem,
  MediaImagesSlide,
  MediasGrid,
  FavoriteItem,
  FavoritesGrid,
};
