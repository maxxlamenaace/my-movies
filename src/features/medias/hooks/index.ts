import { useMediaDetails } from './use-media-details/use-media-details';
import { useMediaGenres } from './use-media-genres/use-media-genres';
import { useAllMedias } from './use-all-medias/use-all-medias';
import { useMediasList } from './use-medias-list/use-medias-list';
import { useFavoriteMedias } from './use-favorite-medias/use-favorite-medias';

export { useMediaDetails, useAllMedias, useMediaGenres, useMediasList, useFavoriteMedias };
