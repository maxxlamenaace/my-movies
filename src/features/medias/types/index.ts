export type MediaType = 'movie' | 'tv';
export type MediaCategory = 'popular' | 'top_rated';
export type MediaImageType = 'poster' | 'backdrop';

export type Media = {
  id: number;
  poster_path?: string;
  adult: boolean;
  overview: string;
  genre_ids: number[];
  original_title: string;
  original_language: string;
  media_type?: MediaType;
  title?: string; // Only for movies
  name?: string; // Only for series
  backdrop_path?: string;
  popularity: number;
  vote_count: number;
  video: boolean;
  vote_average: number;
  release_date?: string; // Only for movies
  first_air_date?: string; // Only for series
};

export type MediaDetails = {
  id: number;
  budget: number;
  genres: Genre[];
  homepage: string;
  revenue: number;
  runtime: number;
  status: string;
  tagline: string;
} & Media;

export type MediaCredit = {
  id: number;
  adult: boolean;
  gender: number;
  known_for_department: string;
  name: string;
  original_name: string;
  popularity: number;
  profile_path: string;
  cast_id: number;
  character: string;
  credit_id: string;
  order: number;
};

export type MediaVideo = {
  id: number;
  type: string;
  official: boolean;
  name: string;
  site: string;
  size: number;
  published_at: Date;
  key: string;
};

export type MediaImage = {
  aspect_ratio: number;
  file_path: string;
  height: number;
  width: number;
};

/* export type ProductionCompany = {
  id: number;
  logo_path: string;
  name: string;
  origin_country: string;
};

export type ProductionCountry = {
  iso_3166_1: string;
  name: string;
};

export type SpokenLanguage = {
  iso_639_1: string;
  name: string;
}; */

export type Genre = {
  id: number;
  name: string;
};
