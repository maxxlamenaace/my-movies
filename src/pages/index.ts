import HomePage from './home/HomePage';
import ActorDetailsPage from './actor-details/ActorDetailsPage';
import SearchPage from './search/SearchPage';
import MediasListPage from './medias-list/MediasListPage';
import MediaDetailsPage from './media-details/MediaDetailsPage';
import FavoritesPage from './favorites/FavoritesPage';

export { HomePage, ActorDetailsPage, SearchPage, MediasListPage, MediaDetailsPage, FavoritesPage };
