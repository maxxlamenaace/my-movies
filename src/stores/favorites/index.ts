export * from './types';
export * from './favorites';
