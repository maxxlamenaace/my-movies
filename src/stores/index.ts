import { useAppStore } from './app/app';
import { useThemeModeStore } from './theme-mode/theme-mode';
import { useFavoritesStore } from './favorites/favorites';

export { useAppStore, useFavoritesStore, useThemeModeStore };
