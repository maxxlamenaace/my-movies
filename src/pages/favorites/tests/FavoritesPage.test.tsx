jest.mock('@/features/medias/hooks', () => ({
  ...jest.requireActual('@/features/medias/hooks'),
  useFavoriteMedias: jest.fn(),
}));

import React from 'react';

import { BrowserRouter } from 'react-router-dom';
import { render, cleanup, waitFor, act, getByText, renderHook } from '@testing-library/react';

import { useFavoriteMedias } from '@/features/medias';
import { mediaDetails } from '@/tests/factories';
import { useFavoritesStore } from '@/stores';
import { FavoritesPage } from '@/pages';

describe('FavoritesPage', () => {
  const useFavoriteMediasMock = jest.mocked(useFavoriteMedias, true);

  afterEach(() => {
    cleanup();
  });

  it('renders correctly', () => {
    act(() => {
      useFavoriteMediasMock.mockImplementation(() => ({
        loading: false,
        favoritesDetails: [],
      }));
    });

    const { asFragment } = render(<FavoritesPage />);

    waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('when favorites array is empty', () => {
    it('renders an error message', () => {
      const { result } = renderHook(() => useFavoritesStore());

      act(() => {
        result.current.favorites = [];
        useFavoriteMediasMock.mockImplementation(() => ({
          loading: false,
          favoritesDetails: [],
        }));
      });

      const { container } = render(<FavoritesPage />);

      waitFor(() => {
        expect(getByText(container, 'No favorites')).toBeDefined();
        expect(container.querySelector('.favorite-grid')).toBeUndefined();
      });
    });
  });

  describe('when favorites array is not empty', () => {
    it('renders with a favorite grid', () => {
      const favoriteMedia = mediaDetails();

      const { result } = renderHook(() => useFavoritesStore());

      act(() => {
        result.current.favorites = [{ mediaId: favoriteMedia.id, mediaType: 'movie' }];
        useFavoriteMediasMock.mockImplementation(() => ({
          loading: false,
          favoritesDetails: [favoriteMedia],
        }));
      });

      const { container } = render(<FavoritesPage />, { wrapper: BrowserRouter });

      waitFor(() => {
        expect(container.querySelector('.favorite-grid')).toBeDefined();
        expect(getByText(container, 'No favorites')).toBeUndefined();
      });
    });
  });
});
