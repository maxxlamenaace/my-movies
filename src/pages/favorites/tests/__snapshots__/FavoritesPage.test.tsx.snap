// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesPage renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-ftnbiv"
  >
    <div
      class="MuiBox-root css-rrfpwj"
    >
      <div
        class="css-9kkbhs-MuiStack-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-1313egv-MuiTypography-root"
        >
          No favorites
        </h4>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          You can add medias to your favorites and you will find all of them here.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
